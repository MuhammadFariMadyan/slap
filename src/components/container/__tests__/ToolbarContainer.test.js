import React from "react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import "jest-styled-components";

import Connected, { ToolbarContainer } from "../ToolbarContainer";

jest.mock("../../presentational/Toolbar", () => "Toolbar");

describe("ToolbarContainer", () => {
  const mockStore = configureMockStore();

  const setup = propOverrides => {
    const props = {
      activeContext: "arrow",
      onClickAddSlide: jest.fn(),
      onClickCursor: jest.fn(),
      onClickTextbox: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<ToolbarContainer {...props} />);

    return {
      wrapper,
      props
    };
  };

  const setupConnected = () => {
    const store = mockStore({
      activeContext: "arrow",
      activeProjectID: "project-1"
    });

    const wrapper = shallow(<Connected store={store} />);

    return {
      wrapper,
      store
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders as connected", () => {
    const { wrapper } = setupConnected();

    expect(wrapper).toMatchSnapshot();
  });

  it("should map new to props", () => {
    const { wrapper } = setupConnected();

    expect(wrapper.props()).toMatchObject({
      activeContext: expect.any(String),
      onClickAddSlide: expect.any(Function),
      onClickCursor: expect.any(Function),
      onClickTextbox: expect.any(Function)
    });
  });

  it("should calls 'onClickAddSlide' props", () => {
    const { wrapper, store } = setupConnected();

    wrapper.props().onClickAddSlide();

    expect(store.getActions()).toEqual([
      {
        type: "ADD_SLIDE",
        payload: expect.any(Object)
      },
      {
        type: "SET_ACTIVE_SLIDE_ID",
        payload: expect.any(String)
      }
    ]);
  });

  it("should calls 'onClickCursor' props ", () => {
    const { wrapper, store } = setupConnected();

    wrapper.props().onClickCursor();

    expect(store.getActions()).toEqual([
      {
        type: "SET_ACTIVE_CONTEXT",
        payload: "arrow"
      },
      {
        type: "SET_ACTIVE_CONTENT_ID",
        payload: null
      },
      {
        type: "SET_IS_CONTENT_EDITABLE",
        payload: false
      }
    ]);
  });

  it("should calls 'onClickTextbox' props ", () => {
    const { wrapper, store } = setupConnected();

    wrapper.props().onClickTextbox();

    expect(store.getActions()).toEqual([
      {
        type: "SET_ACTIVE_CONTEXT",
        payload: "textbox"
      },
      {
        type: "SET_ACTIVE_CONTENT_ID",
        payload: null
      },
      {
        type: "SET_IS_CONTENT_EDITABLE",
        payload: false
      }
    ]);
  });
});
