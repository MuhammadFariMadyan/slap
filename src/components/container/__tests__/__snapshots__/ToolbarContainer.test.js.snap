// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarContainer should renders as connected 1`] = `
<Component
  activeContext="arrow"
  onClickAddSlide={[Function]}
  onClickContext={[Function]}
  onClickCursor={[Function]}
  onClickTextbox={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`ToolbarContainer should renders default 1`] = `
<Toolbar
  items={
    Array [
      Object {
        "icon": <Styled(Component) />,
        "isActive": false,
        "onClick": [Function],
      },
      Object {
        "icon": <Styled(Component) />,
        "isActive": true,
        "onClick": [Function],
      },
      Object {
        "icon": <Styled(Component) />,
        "isActive": false,
        "onClick": [Function],
      },
    ]
  }
/>
`;
